### PR TITLE
Local function support

### LINKED ISSUE #155: Provide signature for local functions
Currently, we provide signature for functions defined in packages and global namespace. We could use XML parse data to also provide signature for locally defined functions (aka closures).

```r
foo <- function(var1) {
  bar <- function(var2) {
    var1 + var2
  }
  bar(|)
}
```

where cursor `|` is at `bar(|)`, the signature of `bar` can be extracted from XML parse data and provided to user.

### PR DESCRIPTION
**Pull request by renkun-ken**
This PR implements local function support for the following providers:

- [x] Signature (Close #155)
- [x] Hover on function name
- [x] Hover on function argument
- [x] Argument completion
- [x] Scope completion documentation
- [x] Test cases

Screencasts

* Signature

image

* Hover on function name

image

* Hover on function argument

image

* Argument completion

image

* Scope completion documentation

image

Kapture 2020-08-28 at 23 11 43

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ jobs:
       run: |
         Rscript -e "install.packages(c('remotes', 'rcmdcheck'), repos = 'https://cloud.r-project.org')"
         Rscript -e "remotes::install_deps(dependencies = TRUE)"
+        Rscript -e "remotes::install_local()"
```
Comment (randy3k) [inline .github/workflows/main.yml:31]: I think that `rcmdcheck::rcmdcheck` installs the package. Why do we need it now?

Comment (renkun-ken) [reply .github/workflows/main.yml]: Sorry, not necessary here indeed.

**Comment by renkun-ken** [comment]
Any other suggestions?

**Comment by renkun-ken** [comment]
Thanks!

I just create some test cases and use the latest version in my work for several days. Everything works as expected.

**Pull request merged**